libtiledbvcf/src/read/reader.cc

```diff
@@ -1427,6 +1427,17 @@ void Reader::info_attribute_count(int32_t* count) {
   *count = this->dataset_->info_field_types().size();
 }
 
+void Reader::sample_count(int32_t* count) {
+  if (count == nullptr)
+    throw std::runtime_error(
+        "Error getting sample count; dataset is not open.");
+  *count = dataset_->metadata().sample_names.size();
+}
+
+void Reader::sample_name(int32_t index, char** name) {
+  *name = const_cast<char*>(dataset_->metadata().sample_names[index].c_str());
```

Comment:
Rather than casting off the `const`, how about making the argument `const char** name`. This is probably the right thing to do anyway because we don't want the caller mutating it since they don't own it.